src/redlock/Redlock.hx

```diff
@@ -0,0 +1,29 @@
+package redlock;
+
+typedef Err = Dynamic;
+typedef LockCallback = Err->Lock->Void;
+
+typedef Options = {
+  // the expected clock drift; for more details see http://redis.io/topics/distlock
+  var driftFactor:Float;
+
+  // the max number of times Redlock will attempt to lock a resource before erroring
+  var retryCount:Int;
+
+  // the time in ms between attempts
+  var retryDelay:Int;
+}
+
+extern class Lock {
+  public function unlock(cb:Void->Void) : Void;
+  public function extend(ttl:Int, cb:LockCallback) : Void;
+}
+
+@:jsRequire('redlock')
+extern class Redlock {
+  public function new(redisClients:Array<Dynamic>, options:Options);
```

Comment:
It felt gross to import redis.Redis inside this file, but maybe we're OK with that?